src/app/static/src/js/components/runWorkflow/runWorkflowReport.vue

```diff
@@ -90,38 +137,172 @@ export default Vue.extend<Data, Methods, Computed, Props>({
+            return this.workflowMetadata.reports.map(r => r.params ? mapRecordToParameterArray(r.params) : []);
+        },
+        stepIsValid: function() {
+            return (this.reportsValid.length > 0) && (this.reportsValid.every(v => v));
         }
     },
     methods: {
-        validateStep: function () {
-            /**
-             *  Valid step should be set to true or false once validation is complete
-             */
-            this.$emit("valid", true)
-        },
         branchSelected(git_branch: string) {
             this.$emit("update", {git_branch});
         },
         commitSelected(git_commit: string) {
             this.$emit("update", {git_commit})
         },
-        updateReports(reports) {
+        getParametersApiCall(report: string) {
+            const commit = this.workflowMetadata.git_commit ? `?commit=${this.workflowMetadata.git_commit}` : '';
+            return api.get(`/report/${report}/config/parameters/${commit}`);
+        },
+        async updateAvailableReports(reports: ReportWithDate[]) {
             this.reports = reports;
-            console.log(`Updating reports with ${JSON.stringify(reports)}`); //TODO: Update reports list
+
+            // We may now have an invalid workflow - it may contain reports or parameters not in the newly selected commit
+            // - remove obsolete reports or params and notify user
+            // 1. Check reports
+            const removals: string[] = [];
+            let newParamsAdded = false;
+            const addToRemovals = (s: string) => {
+                //avoid duplicates
+                if (!removals.includes(s)) {
+                    removals.push(s);
+                }
+            };
+
+            const newReports = [];
+            const availableReportNames = reports.map(r => r.name);
+            const validityIndexRemovals = [];
+            this.workflowMetadata.reports.forEach((r: WorkflowReportWithParams, index: number) => {
+                if (availableReportNames.includes(r.name)) {
+                    newReports.push({...r, params: {...r.params}});
+                } else {
+                    addToRemovals(`Report '${r.name}'`);
+                    validityIndexRemovals.push(index);
+                }
+            });
+
+            //Deal with removing valid flags by index in reverse to avoid moving index bug
+            for(let i = validityIndexRemovals.length-1; i >=0; i--) {
+                this.reportsValid.splice(validityIndexRemovals[i], 1);
+            }
+
+            // 2. Check parameters
+            const calls = newReports.map((r, index) => {
+                return this.getParametersApiCall(r.name)
+                    .then(({data}) => {
+                        const newParameterValues = mapParameterArrayToRecord(data.data);
+                        // Check for parameters in metadata not in fetched params
+                        for (const p of Object.keys(r.params)) {
```

Comment:
for (const **param** of Object.keys(r.params)) 

I think using a letter as a variable name makes the code hard to read.